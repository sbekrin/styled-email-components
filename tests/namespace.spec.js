import React from 'react';
import { renderToStaticMarkup } from 'react-dom/server';
import prettier from 'prettier';
import styled from 'styled-email-components';

const format = html => prettier.format(html, { parser: 'html' });

describe('namespace', () => {
  it('supports vml, wml and office namespaces', () => {
    const RoundRect = styled.vml.roundrect.attrs({
      arcsize: '10%',
      strokecolor: '#1e3650',
      fill: 'true',
    })`
      v-text-anchor: middle;
      height: 40px;
      width: 200px;
    `;
    const Lock = styled.wml.anchorlock``;
    const Fill = styled.vml.fill.attrs({
      type: 'tile',
      src: 'https://i.imgur.com/0xPEf.gif',
      color: '#556270',
    })``;
    const Text = styled.center`
      font-family: sans-serif;
      font-weight: bold;
      font-size: 13px;
      color: #fff;
    `;
    const Button = ({ children, ...rest }) => (
      <RoundRect {...rest}>
        <Lock />
        <Fill />
        <Text>{children}</Text>
      </RoundRect>
    );
    expect(
      format(
        renderToStaticMarkup(
          <Button href="https://example.com">Click me</Button>
        )
      )
    ).toMatchSnapshot(`
"
<v:roundrect arcsize=\\"10%\\"
  strokecolor=\\"#1e3650\\"
  fill=\\"true\\"
  href=\\"https://example.com\\"
  style=\\"v-text-anchor:middle;height:40px;width:200px\\">
  <w:anchorlock />

  <v:fill type=\\"tile\\"
    src=\\"https://i.imgur.com/0xPEf.gif\\"
    color=\\"#556270\\" />

  <center style=\\"font-family:sans-serif;font-weight:bold;font-size:13px;color:#ffffff\\">Click me
  </center>
</v:roundrect>"
`);
  });
});
