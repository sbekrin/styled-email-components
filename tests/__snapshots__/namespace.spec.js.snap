// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`namespace supports vml, wml and office namespaces: 
"
<v:roundrect arcsize=\\"10%\\"
  strokecolor=\\"#1e3650\\"
  fill=\\"true\\"
  href=\\"https://example.com\\"
  style=\\"v-text-anchor:middle;height:40px;width:200px\\">
  <w:anchorlock />

  <v:fill type=\\"tile\\"
    src=\\"https://i.imgur.com/0xPEf.gif\\"
    color=\\"#556270\\" />

  <center style=\\"font-family:sans-serif;font-weight:bold;font-size:13px;color:#ffffff\\">Click me
  </center>
</v:roundrect>"
 1`] = `
"
<v:roundrect arcsize=\\"10%\\"
  strokecolor=\\"#1e3650\\"
  fill=\\"true\\"
  href=\\"https://example.com\\"
  style=\\"v-text-anchor:middle;height:40px;width:200px\\">
  <w:anchorlock />

  <v:fill type=\\"tile\\"
    src=\\"https://i.imgur.com/0xPEf.gif\\"
    color=\\"#556270\\" />

  <center style=\\"font-family:sans-serif;font-weight:bold;font-size:13px;color:#ffffff\\">Click me
  </center>
</v:roundrect>"
`;
